### ISSUE TITLE
License comments are confusing and possibly self-contradictory.

### ISSUE THREAD

**Issue by simonbuchan** [awslabs/aws-crt-nodejs#367]
For reference, all the source files currently have the following license comment:

```
/*
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0.
 */
```

Now I'm not a lawyer, and [wikipedia](https://en.wikipedia.org/wiki/All_rights_reserved) says that the meaning of "All Rights Reserved." isn't exactly what it seems like, but these two lines are unnecessarily confusing together, and possibly contradict each other. In particular, any community pull requests or other contributions under the Apache license do *not* constitute a transfer of copyright to the licensor (Amazon), merely:

>  any Contribution intentionally submitted for inclusion in the Work by You to the Licensor shall be under the terms and conditions of this License, without any additional terms or conditions.

In other words, every contributor in an open source code base normally retains copyright to whatever contributions they have made and "merely" license them to any users of the code base under the apache (or whatever) terms. In particular, this doesn't grant exclusivity to the contribution: they are free to copy *their own* code to a GPL codebase, use commercially, etc.. I don't think "Amazon.com, Inc. or its affiliates" sufficiently covers this nuance.

For reference, the AWS SDK v3 doesn't bother with license blocks at all, each package and the repository simply include the `LICENSE` file and SPDX id in the package.json, since that's expected and sufficient.

**Comment by bretambrose** [comment]
License file and format is following guidance we received from our open source legal liason(s).  I don't have any contact info for AWS Legal, so if you have a concrete issue or concern you'll probably need to go through customer service.
